src/compiler/Restler.Compiler/Compiler.fs

```diff
@@ -858,15 +858,59 @@ let generateRequestPrimitives (requestId:RequestId)
+                match customPayloadType with
+                | CustomPayloadType.Header ->
+                    dictionary.getCustomPayloadHeaderParameterNames()
+                | CustomPayloadType.Query ->
+                    dictionary.getCustomPayloadQueryParameterNames()
+                | _ ->
+                    raise (invalidArg "customPayloadType" (sprintf "%A is not supported in this context." customPayloadType))
+            
+            let isRequestSpecificName (name:string) = 
+                // Request-specific custom payloads are of the form <endpoint>/<method>/<parameter name>
+                name.StartsWith('/') &&
+                    SupportedOperationMethods |> Array.exists (fun method -> name.ToLower().Contains(sprintf "/%s/" method))
+
+            parametersSpecifiedAsCustomPayloads 
+            |> Seq.filter (fun name -> isRequestSpecificName name && not (name.StartsWith(prefix)))
+            |> Seq.toList
+
+        // Filter out the spec parameters
+        // Filter out Content-Type, because this is handled separately later, in order to be able to fuzz or replace the content type
+        // Filter both the global content-type and the one for this specific request.
+        let excludedContentTypeParameterNamesInCustomPayload =
+            ["Content-Type" ; (dictionary.getRequestTypePayloadName requestId.endpoint (requestId.method.ToString()) "Content-Type")]
+
+        let injectedCustomPayloadParameters = 
+            getInjectedCustomPayloadParameters dictionary customPayloadType
+                                               (specParameters @ excludedContentTypeParameterNamesInCustomPayload @ excludedRequestSpecificParameters)
+        injectedCustomPayloadParameters
+
     // Generate header parameters.
     // Do not compute dependencies for header parameters unless resolveHeaderDependencies (from config) is true.
     let requestHeaderParameters =
         let headersSpecifiedAsCustomPayloads = dictionary.getCustomPayloadHeaderParameterNames()
         match requestParameters.header with
         | [] ->
-            // Filter out Content-Type, because this is handled separately below
-            let injectedCustomPayloadHeaderParameters = getInjectedCustomPayloadParameters dictionary CustomPayloadType.Header ["Content-Type"]
+            // No headers in Swagger
+            let injectedCustomPayloadHeaderParameters = getInjectedHeaderOrQueryParameters [] CustomPayloadType.Header
             [(ParameterPayloadSource.DictionaryCustomPayload, ParameterList (injectedCustomPayloadHeaderParameters |> seq))]
+        | [(Schema, ParameterList y)] when y |> Seq.isEmpty ->
+            // No headers in Swagger
+            let injectedCustomPayloadHeaderParameters = getInjectedHeaderOrQueryParameters [] CustomPayloadType.Header
+            if injectedCustomPayloadHeaderParameters |> Seq.isEmpty then
+                // TODO: remove this branch and update grammar.py test baselines
```

Comment:
Just wanted to check if this TODO is for this PR, or if it's to be addressed in future set of changes.

Reply:
It's for a future set of changes, updating the grammar.json files is a bit time consuming - in addition to the compiler updates, need to make sure the engine can consume the new grammar.json files and this particular change could introduce a regression.